modules/custom/mentions/src/Plugin/Mentions/Entity.php

```diff
@@ -66,8 +95,11 @@ public function outputCallback($mention, $settings) {
   public function targetCallback($value, $settings) {
     $entity_type = $settings['entity_type'];
     $input_value = $settings['value'];
-    $query = $this->entityQueryService->get($entity_type);
-    $result = $query->condition($input_value, $value)->execute();
+    $query = $this->entityQueryService
+      ->get($entity_type)
+      ->condition($input_value, $value)
+      ->accessCheck(FALSE);
```

Comment:
this should probably not be here

Reply:
@ronaldtebrake any suggestions on this?

Reply:
I have been looking at this morning. I think we cannot get past by this accesCheck false.

The targetCallback is being used on quite some places though, not just the cron. So to make it any safer I can imagine we set a variable or such in the the activity_send_email_cron so when the functions gets run from the cron it will disable the access check only during the cron.

@ronaldtebrake @navneet0693 what do you think? As anonymous users are not allowed to view profiles I don't think we can remove this check in total.

Reply:
Hey @robertragas and @ronaldtebrake ,

I see some movement in this PR. I'm also working on the `social_profile` module which as one of the changes will give anonymous users access to profile information (which will then be checked on a per-field level depending on configuration).

So this may already solve this problem without removing access checks?